### PR TITLE
Add discovery functionality

### PR DESCRIPTION
**Pull request by ManasaRaoK**
Add GET functionality on /discovery endpoint to retrieve a list of devices discovered by SLP discovery.

### CODE REVIEW

**Comment by walteraa** [comment]
@CharlleDaniel could you review it, please?

File: spec/xclarity_client_discovery_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @excludeAttributes = %w(lastUpdateElapsedTime discoveryInProgress)
+  end
+
+  it 'has a version number' do
+    expect(XClarityClient::VERSION).not_to be nil
+  end
+
+  describe 'GET /discovery' do
+    it 'should respond with an array' do
+      expect(@client.discover_devices_by_slp).not_to be_empty
```
Comment (CharlleDaniel) [inline spec/xclarity_client_discovery_spec.rb]: Hi @ManasaRaoK, in the test below, I think that you are not testing if the result is an array
```
 it 'should respond with an array' do
￼   expect(@client.discover_devices_by_slp).not_to be_empty
end
```

I prefer that you use in instead
```
    it 'should respond with an array' do
      expect(@client.discover_devices_by_slp.class).to eq(Array)
    end
```

File: spec/xclarity_client_discovery_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    :auth_type => ENV['LXCA_AUTH_TYPE'],
+    :verify_ssl => ENV['LXCA_VERIFY_SSL']
+    )
+
+    @client = XClarityClient::Client.new(conf)
+
+    @host = ENV['LXCA_HOST']
+  end
+
+  before :each do
```
Comment (CharlleDaniel) [inline spec/xclarity_client_discovery_spec.rb]: Just a question, why you added these lines?

```
before :each do
     @includeAttributes = %w(lastUpdateElapsedTime discoveryInProgress)
     @excludeAttributes = %w(lastUpdateElapsedTime discoveryInProgress)
  end
```
I think that isn't being used.

File: spec/xclarity_client_discovery_spec.rb (hunk 1 above)
Comment (ManasaRaoK) [reply spec/xclarity_client_discovery_spec.rb]: Thanks for catching this. Addressed now.

File: spec/xclarity_client_discovery_spec.rb (hunk 2 above)
Comment (ManasaRaoK) [reply spec/xclarity_client_discovery_spec.rb]: Thanks for catching this. Addressed now.

**Comment by AndreyMenezes** [comment]
@ManasaRaoK do a rebase here. Do you have four commits for one simply functionality.

**Comment by CharlleDaniel** [comment]
@walteraa the code looks good for me, can you merge this pr?

**Comment by chargio** [comment]
@ManasaRaoK Is this for Gaprisdanvhili or not? I don't see the tag

**Pull request merged**